File: ios/BirikimYap/Siri/SiriBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SiriBridge, NSObject)

RCT_EXTERN_METHOD(getPendingSiriExpenses:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(clearPendingSiriExpenses:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/BirikimYap/Siri/SiriBridge.swift
import Foundation
import React

@objc(SiriBridge)
class SiriBridge: NSObject {
    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }

    @objc
    func getPendingSiriExpenses(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        let userDefaults = UserDefaults(suiteName: "group.com.birikimyapsiri.app") ?? UserDefaults.standard
        let pending = userDefaults.array(forKey: "pending_siri_expenses") as? [[String: Any]] ?? []
        resolve(pending)
    }

    @objc
    func clearPendingSiriExpenses(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        let userDefaults = UserDefaults(suiteName: "group.com.birikimyapsiri.app") ?? UserDefaults.standard
        userDefaults.removeObject(forKey: "pending_siri_expenses")
        userDefaults.synchronize()
        resolve(true)
    }
}
